src/sage/crypto/public_key/key_exchange/key_exchange.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+from sage.structure.sage_object import SageObject
+
+class KeyExchange(SageObject, ABC):
+
+    @abstractmethod
+    def alice_secret_key(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def alice_public_key(self, alice_secret_key):
+        raise NotImplementedError
+
+    @abstractmethod
+    def bob_secret_key(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def bob_public_key(self, bob_secret_key):
+        raise NotImplementedError
+
+    @abstractmethod
+    def alice_compute_shared_secret(self, alice_sk, bob_pk):
+        raise NotImplementedError
+
+    @abstractmethod
+    def bob_compute_shared_secret(self, bob_sk, alice_pk):
+        raise NotImplementedError
+
+    def alice_keygen(self):
```

Comment:
```suggestion
    def alice_keygen(self) -> tuple[Any, Any]:
```

And add `from typing import Any`